.github/workflows/github-schema-validation.yml

```diff
@@ -0,0 +1,48 @@
+name: '.github schema check'
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+    paths:
+      - '.github/dependabot.yml'
+      - '.github/actions/*/action.yml'
+      - '.github/workflows/*.yml'
+  pull_request:
+    paths:
+      - '.github/dependabot.yml'
+      - '.github/actions/*/action.yml'
+      - '.github/workflows/*.yml'
+
+jobs:
+  github-schemas:
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4.2.1
+        with:
+          fetch-depth: 1
+
+      - name: '.github/dependabot.yml'
+        if: ${{ hashFiles('.github/dependabot.yml') != '' }}
+        uses: cardinalby/schema-validator-action@v3
```

Comment:
I guess, but I'd prefer to have a command line script one could run locally, instead of having an extra third party dependency (on the action) and needing to push to GH for the validation to work.